lib/ansible/plugins/inventory/ovirt.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Red Hat, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: ovirt
+    plugin_type: inventory
+    short_description: oVirt inventory source
+    version_added: "2.10"
+    author: Bram Verschueren (@bverschueren)
+    requirements:
+      - ovirt-engine-sdk-python >= 4.2.4
+    extends_documentation_fragment:
+        - inventory_cache
+        - constructed
+    description:
+      - Get inventory hosts from the ovirt service.
+      - Requires a YAML file ending in 'ovirt.yml', 'ovirt4.yml', 'ovirt.yaml', 'ovirt4.yaml'.
+    options:
+      plugin:
+        description: the name of this plugin, it should always be set to 'ovirt' for this plugin to recognise it as it's own.
+        required: True
+        choices: ['ovirt']
+      ovirt_url:
+        description: URL to ovirt-engine API.
+        required: True
+        env:
+          - name: OVIRT_URL
+      ovirt_username:
+        description: ovirt authentication user.
+        required: True
+        env:
+          - name: OVIRT_USERNAME
+      ovirt_password:
+        description: ovirt authentication password.
+        required : True
+        env:
+          - name: OVIRT_PASSWORD
+      ovirt_cafile:
+        description: path to ovirt-engine CA file.
+        required: False
+      ovirt_query_filter:
+        required: false
+        description: dictionary of filter key-values to query VM's. See U(https://ovirt.github.io/ovirt-engine-sdk/master/services.m.html#ovirtsdk4\
+.services.VmsService.list) for filter parameters.
+      ovirt_hostname_preference:
+        required: false
+        description: list of options that describe the ordering for which hostnames should be assigned. See U(https://ovirt.github.io/ovirt-engin\
+e-api-model/master/#types/vm) for available attributes.
+        default: ['fqdn', 'name']
+        type: list
+'''
+
+EXAMPLES = '''
+# Sample content of ovirt.yml
+# plugin: ovirt
+# ovirt_url: http://localhost/ovirt-engine/api
+# ovirt_username: ansible-tester
+# ovirt_password: secure
+# ovirt_query_filter:
+#   search: 'name=myvm AND cluster=mycluster'
+#   case_sensitive: no
+#   max: 15
+# keyed_groups:
+#   - key: cluster
+#     prefix: 'cluster'
+# groups:
+#   dev: "'dev' in tags"
+# compose:
+#   ansible_host: devices["eth0"][0]
```

Comment:
Just an idea but it would be great to explain how the `keyed_groups` and `groups` works and what it does (I might miss some docs but I have not found any good explanations)

Reply:
This is explained here: https://docs.ansible.com/ansible/latest/plugins/inventory.html#using-inventory-plugins